cblox/include/cblox/core/submap.h

```diff
@@ -0,0 +1,52 @@
+#ifndef CBLOX_CORE_SUBMAP_H_
+#define CBLOX_CORE_SUBMAP_H_
+
+#include <mutex>
+
+#include "cblox/core/common.h"
+
+namespace cblox {
+
+// Pure virtual class representing a submap.
+class Submap {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  // Constructor.
+  Submap(const Transformation& T_M_S, SubmapID submap_id)
+      : submap_id_(submap_id), T_M_S_(T_M_S) {}
+
+  ~Submap() {}
+
+  // Submap pose interaction.
+  inline const Transformation& getPose() const {
+    std::unique_lock<std::mutex> lock(transformation_mutex_);
+    return T_M_S_;
+  }
+
+  inline void setPose(const Transformation& T_M_S) {
+    std::unique_lock<std::mutex> lock(transformation_mutex_);
+    T_M_S_ = T_M_S;
+  }
+
+  inline SubmapID getID() const { return submap_id_; }
+
+  virtual size_t getNumberAllocatedBlocks() const = 0;
+
+  virtual size_t getMemorySize() const = 0;
+
+  virtual void finishSubmap() = 0;
+
+  virtual void prepareForPublish() = 0;
+
+ protected:
+  SubmapID submap_id_;
```

Comment:
Shall we make this const? I guess the ID should be "baked into" the object at construction time.